zcash_client_sqlite/src/wallet/common.rs

```diff
@@ -158,9 +162,16 @@ where
                  AND recipient_key_scope IS NOT NULL
                  AND nf IS NOT NULL
                  AND commitment_tree_position IS NOT NULL
-                 AND spent IS NULL
                  AND transactions.block <= :anchor_height
                  AND {table_prefix}_received_notes.id NOT IN rarray(:exclude)
+                 AND {table_prefix}_received_notes.id NOT IN (
+                   SELECT {table_prefix}_received_note_id
+                   FROM {table_prefix}_received_note_spends
+                   JOIN transactions stx ON stx.id_tx = transaction_id
+                   WHERE stx.block IS NOT NULL -- the spending tx is mined
+                   OR stx.expiry_height IS NULL -- the spending tx will not expire
+                   OR stx.expiry_height > :anchor_height -- the spending tx is unexpired
```

Comment:
Why are we comparing expiry height to anchor height? Expiry height is only relevant to when a transaction can be mined.